hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMSecurityProtocolServer.java

```diff
@@ -403,13 +415,20 @@ public List<String> listCACertificate() throws IOException {
   }
 
   @Override
-  public String getRootCACertificate() throws IOException {
-    LOGGER.debug("Getting Root CA certificate.");
-    if (storageContainerManager.getScmStorageConfig()
-        .checkPrimarySCMIdInitialized()) {
-      return CertificateCodec.getPEMEncodedString(rootCACertificate);
+  public synchronized String getRootCACertificate() throws IOException {
```

Comment:
Do we need synchronized here?, The old existing method is without synchronized.

Reply:
Yes, since the list of the root cas can change. The old method didn't need synch because the whole server could only ever use one root ca